pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/MaxAggregationFunction.java

```diff
@@ -73,10 +73,11 @@ public void aggregate(int length, AggregationResultHolder aggregationResultHolde
     BlockValSet blockValSet = blockValSetMap.get(_expression);
     if (_nullHandlingEnabled) {
       RoaringBitmap nullBitmap = blockValSet.getNullBitmap();
-      if (nullBitmap != null && !nullBitmap.isEmpty()) {
-        aggregateNullHandlingEnabled(length, aggregationResultHolder, blockValSet, nullBitmap);
-        return;
+      if (nullBitmap == null) {
```

Comment:
Ideally we still want to avoid the null bitmap check when it is `null` or empty for better performance. We can merge this fix for now, but let's add a TODO to track it. Same for other functions that uses double result holder